src/t_stream.c

```diff
@@ -1338,6 +1345,132 @@ robj *createObjectFromStreamID(streamID *id) {
                         id->ms,id->seq));
 }
 
+/* Get the first or last entry ID from a non-empty stream. */
+void streamGetTipID(stream *s, streamID *id, int first) {
+    streamIterator si;
+    int64_t numfields;
+    streamID min, max;
+    min.seq = 0;
+    min.ms = 0;
+    max.seq = UINT64_MAX;
+    max.seq = UINT64_MAX;
+
+    streamIteratorStart(&si,s,&min,&max,!first);
+    streamIteratorGetID(&si,id,&numfields);
+    streamIteratorStop(&si);
+}
+
+/* Returns 1 if the ID is 0-0. */
+int streamIDEqZero(streamID *id) {
+    return !(id->ms || id->seq);
+}
+
+/* Check if a range contains an XDEL.
+ * Assumes start < end && end < s->last_id. */
+int streamIsContiguousRange(stream *s, streamID *start, streamID *end) {
+    streamID start_id, end_id;
```

Comment:
end_id is unused.. bug?

also, what happens if the stream has entries 1 thru 10, when 3 and 7 have been XDELeted
and i call this function with start=1, end=6? it'll return 1 even though the range isn't continuous

Reply:
You're very correct, end_id is unused - fixed.

After the fix the "what happens" will not happen.